gmail_fetcher.py

```
@@ -154,13 +101,74 @@ def __init__(self, email_address: str, app_password: str):
         """Establish connection to Gmail IMAP server."""
         try:
+            # Use original credentials without encoding
+            email = self.email_address
+            password = self.password
+            
             self.conn = imaplib.IMAP4_SSL(self.imap_server)
-            self.conn.login(self.email_address, self.password)
-        except Exception as e:
+            self.conn.login(email, password)
+        except imaplib.IMAP4_SSL.error as e:
             raise Exception(f"Failed to connect to Gmail: {str(e)}")
```

Ruff (0.7.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)